lib/browse_everything/driver/file_system.rb

```diff
@@ -36,18 +44,32 @@ def details(path, display = nil)
         return nil unless File.exist?(path)
         info = File::Stat.new(path)
         BrowseEverything::FileEntry.new(
-          Pathname.new(File.expand_path(path)).relative_path_from(Pathname.new(config[:home])),
+          make_pathname(path),
           [key, path].join(':'),
-          display || File.basename(path),
+          make_display(path, display),
           info.size,
           info.mtime,
           info.directory?
         )
       end
 
+      def make_pathname(path)
+        Pathname.new(File.expand_path(path)).relative_path_from(Pathname.new(config[:home]))
+      end
+
+      def get_file_size(path)
+        File.size(path).to_i
+      rescue
```

Comment:
I don't you suppose you know what exception the original author was checking for do you?